packages/core/src/sequelize.d.ts

```diff
@@ -172,7 +172,7 @@ export interface Config {
   readonly dialectOptions: Readonly<DialectOptions>;
 }
 
-export type Dialect = 'mysql' | 'postgres' | 'sqlite' | 'mariadb' | 'mssql' | 'db2' | 'snowflake' | 'ibmi';
+export type Dialect = 'mysql' | 'postgres' | 'sqlite' | 'mariadb' | 'mssql' | 'db2' | 'snowflake' | 'ibmi' | 'cockroachdb';
```

Comment:
@ephys On adding `cockroachdb` here I'm getting a `TS2590: Expression produces a union type that is too complex to represent.` error. I ended up circumventing this by temporarily adding `type ExpectationKey = 'default' | 'cockroachdb' | Permutations<Exclude<Dialect, 'cockroachdb'>>;` But if possible I would like to know why we need the Permutations here to find a better solution!

Reply:
I'll need to fix that in its own PR. I'll need to replace permutations with something like sorted permutations, I'll look into that

Reply:
The Permutations are used in the unit tests, for the `expectsql` function so we can define various combinations of dialects that we have the same expectations for. I think that's still something that we want to have so we should see if we can find a different way to get the same result

Reply:
We don't need to accept all possible permutations, we can have their order sorted to drastically reduce the number of possible combinations

Reply:
Update: trying to sort wouldn't help, but putting a depth limiter would work

Reply:
Depth limiter should work. If something has more than half of the dialects you should use `default`. We can compute that number of dialects so we don't have to manually update it when adding additional dialects

Reply:
@WikiRik Would something like this work ?
```
type Decrement<N extends number> = N extends 0
  ? 0
  : N extends 1
  ? 0
  : N extends 2
  ? 1
  : N extends 3
  ? 2
  : N extends 4
  ? 3
  : N extends 5
  ? 4
  : N extends 6
  ? 5
  : N extends 7
  ? 6
  : N extends 8
  ? 7
  : N extends 9
  ? 8
  : N extends 10
  ? 9
  : never;

const depth =  4;

type Permutation<T extends string, D extends number, U extends string = T> = T extends any ?  D extends 0 ? 'default' : T | `${T} ${Permutation<Exclude<U, T>, Decrement<D>>}` : never;
```

Still not sure how I can detect the length of the members inside the type Dialect though.

Reply:
It's here: https://github.com/sequelize/sequelize/pull/15595

but I need to take a minute to apply the changes from the last review

Reply:
That PR has been merged so you should be able to add `cockroachdb` here without issues

Reply:
Amazing! thanks